app/app/VotingList.php

```diff
@@ -59,4 +60,15 @@ public function getFormattedDateAttribute()
     {
         return $this->date->isoFormat('dddd, MMMM D, YYYY');
     }
+
+    public function sharePictureUrl(): string
+    {
+        if ($this->vote?->isPrimaryVote()) {
+            if (Storage::disk('public')->exists("share-pictures/vote-sharepic-{$this->id}.png")) {
+                return Storage::disk('public')->url("share-pictures/vote-sharepic-{$this->id}.png");
+            }
+        }
+
+        return '';
+    }
```

Comment:
You need a nullable return type, as a share picture URL might not exists. Apart from that, I’d go for early returns instead of nested ifs.

```suggestion
    public function sharePictureUrl(): ?string
    {
        if (! $this->vote?->isPrimaryVote()) {
          return null;
        }
        
        $path = "share-pictures/vote-sharepic-{$this->id}.png";
        $disk = Storage::disk('public');
        
        if (! $disk->exists($path) {
          return null;
        }
        
        return $disk->url($path);
    }
```